### PR TITLE
PWGHF: add configurables to turn off/on creation of different 3-prong species

### PR DESCRIPTION
**Pull request by fgrosa**
This PR adds the possibility to create the 3-prong candidates only for a subset of the possible particle species. Since in most of the cases we run it only for one species, this should improve both the memory and CPU consumption.

### CODE REVIEW

**Review by mfaggin — COMMENTED** [review]
ciao @fgrosa many thanks for this very useful development, which I agree 100% with.
Just one question from my side: what about switching everything off by default and announcing this change in the Mattermost channel (i.e. that one has to switch on each desired channel) to avoid waste of resources?

File: PWGHF/TableProducer/candidateCreator3Prong.cxx [hunk 1]
```diff
@@ -70,6 +76,12 @@ struct HfCandidateCreator3Prong {
   double massPiKPi{0.};
   double bz{0.};
 
+  using FilteredHf3Prongs = soa::Filtered<aod::Hf3Prongs>;
+  using FilteredPvRefitHf3Prongs = soa::Filtered<soa::Join<aod::Hf3Prongs, aod::HfPvRefit3Prong>>;
+
+  // filter collisions
+  Filter filterSelected3Prongs = (createDplus && (o2::aod::hf_track_index::hfflag & static_cast<uint8_t>(BIT(aod::hf_cand_3prong::DecayType::DplusToPiKPi))) != static_cast<uint8_t>(0)) || (createDs && (o2::aod::hf_track_index::hfflag & static_cast<uint8_t>(BIT(aod::hf_cand_3prong::DecayType::DsToKKPi))) != static_cast<uint8_t>(0)) || (createLc && (o2::aod::hf_track_index::hfflag & static_cast<uint8_t>(BIT(aod::hf_cand_3prong::DecayType::XicToPKPi))) != static_cast<uint8_t>(0)) || (createLc && (o2::aod::hf_track_index::hfflag & static_cast<uint8_t>(BIT(aod::hf_cand_3prong::DecayType::XicToPKPi))) != static_cast<uint8_t>(0));
```
Comment (vkucera) [inline PWGHF/TableProducer/candidateCreator3Prong.cxx:83]: Using `TESTBIT` would make it more readable.
```suggestion
  // filter candidates
  Filter filterSelected3Prongs = (createDplus && (o2::aod::hf_track_index::hfflag & static_cast<uint8_t>(BIT(aod::hf_cand_3prong::DecayType::DplusToPiKPi))) != static_cast<uint8_t>(0)) || (createDs && (o2::aod::hf_track_index::hfflag & static_cast<uint8_t>(BIT(aod::hf_cand_3prong::DecayType::DsToKKPi))) != static_cast<uint8_t>(0)) || (createLc && (o2::aod::hf_track_index::hfflag & static_cast<uint8_t>(BIT(aod::hf_cand_3prong::DecayType::XicToPKPi))) != static_cast<uint8_t>(0)) || (createLc && (o2::aod::hf_track_index::hfflag & static_cast<uint8_t>(BIT(aod::hf_cand_3prong::DecayType::XicToPKPi))) != static_cast<uint8_t>(0));
```

**Review by vkucera — COMMENTED** [review]
As a further improvement in this direction, `HfCandidateCreator3ProngExpressions` could inspect the `create...` configurables in `HfCandidateCreator3Prong` and skip MC matching for disabled species.

**Comment by fgrosa** [comment]
Hi @mfaggin and @vkucera many thanks for your comments, I have implemented all of them except for the one regarding `TESTBIT`, since I get an error that I didn't manage to fix. However, if you agree I would merge it without it (and then if I find a way to make it work in the `Filter`, I will update it in a next PR).

**Review by vkucera — APPROVED** [review]
Looks good to me. Thanks @fgrosa

**Pull request merged**